plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcCachingConnectorSmokeTest.java

```diff
@@ -32,11 +38,60 @@ protected QueryRunner createQueryRunner()
                 .putAll(TestingH2JdbcModule.createProperties())
                 .put("metadata.cache-ttl", "10m")
                 .put("metadata.cache-missing", "true")
+                .put("case-insensitive-name-matching", "true")
                 .put("allow-drop-table", "true")
                 .build();
+        this.h2SqlExecutor = new JdbcSqlExecutor(properties.get("connection-url"), new Properties());
         return createH2QueryRunner(REQUIRED_TPCH_TABLES, properties);
     }
 
+    @Test
+    public void testFlushMetadataCacheProcedureFlushMetadata()
+    {
+        assertUpdate("CREATE TABLE cached AS SELECT * FROM nation", 25);
+        assertQuerySucceeds("SELECT name, regionkey FROM cached");
+
+        // Rename column outside Trino
+        h2SqlExecutor.execute("ALTER TABLE tpch.cached ALTER COLUMN regionkey RENAME TO renamed");
+
+        String renamedColumnQuery = "SELECT name, renamed FROM cached";
+        // Should fail as Trino has old metadata cached
```

Comment:
The caching happens in two separate places:
1. CachingIdentifierMapping (schema/table mappings cache)
2. CachingJdbcClient (schema/table/column/stats cache)

`testFlushMetadataCacheProcedureFlushMetadata` tests the column cache (from CachingJdbcClient) and `testFlushMetadataCacheProcedureFlushIdentifierMapping` tests the table mapping cache but does not test the schema mapping cache.

Can you add an additional test for schema renames? I think [H2 supports schema renames](https://www.h2database.com/html/commands.html#alter_schema_rename)

Reply:
No problem I can add this test.
But I have doubts if it will not be redundant as:
- column and schema cache is inside Jdbc clent. Both are flushed with same/single method call `flushCache`
- both use same caching lib/implementation
- both are fushed with same method `invalidateAll` on cache level in single `flushCache` call

So we would just assure that flushCache method calls all relevant underlining caches.

Let me know your thoughts.

Maybe i will just extend existing test ?
Ofcourse if schema rename will be possible in H2, i will need to check that.

Reply:
I've extended existing test to cover column, table and schema cache.
https://github.com/trinodb/trino/blob/64ea4ea5f55396b0a6f792dfc5842d0631b19643/plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcCachingConnectorSmokeTest.java#L70:L98